firehydrant/rotations_test.go

```diff
@@ -0,0 +1,155 @@
+package firehydrant
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func expectedRotationResponse() *RotationResponse {
+	return &RotationResponse{
+		ID:                              "rotation-id",
+		Name:                            "A pleasant rotation",
+		Description:                     "Managed by Terraform. Contact @platform-eng for changes.",
+		TimeZone:                        "America/New_York",
+		Color:                           "#ff0000",
+		SlackUserGroupID:                "slack-group-1",
+		PreventShiftDeletion:            true,
+		CoverageGapNotificationInterval: "an-appropriate-interval",
+		Strategy: RotationStrategy{
+			Type:        "weekly",
+			HandoffTime: "09:00:00",
+			HandoffDay:  "monday",
+		},
+		Members: []RotationMember{
+			{
+				ID:   "77779528-690b-4161-84ca-312e932c626e",
+				Name: "Frederick Graff",
+			},
+		},
+		Restrictions: []RotationRestriction{
+			{
+				StartDay:  "monday",
+				StartTime: "09:00:00",
+				EndDay:    "friday",
+				EndTime:   "17:00:00",
+			},
+		},
+	}
+}
+
+func expectedRotationResponseJSON() string {
```

Comment:
Nit: if `expectedRotationResponseJSON` and `expectedRotationResponse` need to be in sync, couldn't we marshal the result of `expectedRotationResponse` to JSON?

Reply:
In this case, that's kinda what we're testing, allowing for the possibility that the client is doing something to the json response before it becomes an object.  It's probably not necessary to be this explicit (we don't need to test that json.Marshall() works), but I think it's cleaner to look at and compare a live API response (which I always use when writing these) to the objectified version.  Probably comes from a deep place of mistrust produced by unresolved personal trauma that would make me a great candidate to be some kind of costumed vigilante and yet here I am...

If it bothers you enough to want it changed, say so, but it works as written, so...

Reply:
I'm fine with leaving it this way. The approach makes sense now that I understand what you are intending.

> Probably comes from a deep place of mistrust produced by unresolved personal trauma that would make me a great candidate to be some kind of costumed vigilante and yet here I am...

I look forward to your comic series